StreetCare/StreetCare/Views/Screens/CommunityEventView.swift

```diff
@@ -237,56 +260,71 @@ class SearchCommunityModel: ObservableObject {
             }.filter { !$0.isEmpty }
         }
     }
+    
+        func filterByDate(filterType: FilterType) {
+            let calendar = Calendar.current
+            var filteredEvents = events
 
-    func filterEvents(eventType: EventType) {
```

Comment:
This function cannot be removed. This is the root cause as to why you cannot see events populating in the view. 

```
// When the CommunityEventView loads in the onAppear() call the adapter fetches the events from Firebase
 .onAppear {
            if let user = Auth.auth().currentUser {
                self.user = user
                adapter.delegate = self
                adapter.refresh() // events fetched from Firebase here
            }
        }
// then func eventDataRefreshed(...) observes the data fetched from Firebase and updated viewModel.filterEvents

extension CommunityEventView: EventDataAdapterProtocol {
    func helpRequestDataRefreshed(_ events: [HelpRequest]) {
    }
    
    func eventDataRefreshed(_ events: [Event]) {
        viewModel.events = events.filter({ event in
            return event.eventDate != nil
        })
        viewModel.filterEvents(eventType: eventType)
    }
}

// now this function is called but in this PR the function has been removed therefore no events are populating the view

    func filterEvents(eventType: EventType) {
        if events.isEmpty { return }
        if eventType == .future {
            filterAndSortEvents(eventType: .future, dateComparison: { $0 > $1 }, dayComparison: { $0 < $1 }, monthComparison: { $0 < $1 })
        } else if eventType == .past {
            filterAndSortEvents(eventType: .past, dateComparison: { $0 < $1 }, dayComparison: { $0 > $1 }, monthComparison: { $0 > $1 })
        }
    }